src/server/trpc.ts

```diff
@@ -36,11 +37,25 @@ const t = initTRPC.context<Context>().create({
   },
 })
 
-/**
- * Create a router
- * @see https://trpc.io/docs/v10/router
- */
-export const router = t.router
+// Setting outer context with TPRC will not get us correct path during request batching, only by setting logger context in
+// the middleware do we get the exact path to log
+const loggerMiddleware = t.middleware(async ({ path, next }) => {
+  const start = Date.now()
+  const logger = createBaseLogger(path)
+  const durationInMs = Date.now() - start
```

Comment:
I think this is slightly off -- the duration should only be logged after the `next` call, yes? So the procedure's duration can be accurately captured

Reply:
refactored wrongly and made this mistake, good catch!